src/test/java/com/devonfw/sample/archunit/ArchitectureTest.java

```diff
@@ -15,16 +15,23 @@ public class ArchitectureTest {
 
   @ArchTest
   private static final ArchRule shouldOnlyAccessValidLayers = //
-      layeredArchitecture().consideringAllDependencies() //
-          .layer("common").definedBy("..common..") //
-          .layer("logic").definedBy("..logic..") //
-          .layer("dataaccess").definedBy("..dataaccess..") //
-          .layer("service").definedBy("..service..") //
-          .layer("batch").definedBy("..batch..") //
-          .layer("ui").definedBy("..ui..") //
-          // TODO
-          .because("Dependency of technical layers violates architecture rules.");
+      layeredArchitecture().consideringAllDependencies() //      
+      .layer("common").definedBy("..common..") //
+      .layer("logic").definedBy("..logic..") //
+      .layer("dataaccess").definedBy("..dataaccess..") //
+      .layer("service").definedBy("..service..") //
+      .layer("client").definedBy("..client..")
+      .layer("batch").definedBy("..batch..")
+
+      .whereLayer("client").mayNotBeAccessedByAnyLayer()
+      .whereLayer("batch").mayOnlyBeAccessedByLayers( "logic")
```

Comment:
IMHO code dependencies from `logic` layer to `batch` are violating the layered architecture since the `batch` layer is "above" the `logic` layer just like the `service` layer:
https://github.com/devonfw/sonar-devon4j-plugin/blob/master/documentation/rules.asciidoc#architectural-dependency-rules

Typically batches are started via
* timers (like cron expressions with e.g. quartz - e.g. every 5 minutes, every night at 1:55 or every Friday at 22:00)
* external events - typically data to import was delivered

However, there are applications with architecture where batches are triggered from end-users via use-cases. This is not what devonfw recommends as batches should scale independent from OLTP and also batches have a tendency to cause trouble (performance drain, OutOfMemoryException) and in such case this would bring your OLTP down as well.
Further, as you can see below `batch` may depend on `logic` what introduces a cyclic dependency between layers what is discouraged. Often you simple decouple the triggering of batches from code-invocations as you simple write specific status or records into the DB that makes a batch to process that data.

To summarize, I would suggest to use `mayNotBeAccessedByAnyLayer()` as well for `batch`.

Reply:
Thank you for your elaborate explanation. I appreciate it.

Reply:
Thanks for your update and rework.